### PR TITLE
Test CMake project with CI

### PR DESCRIPTION
**Pull request by o01eg**
Includes https://github.com/godotengine/godot-cpp/pull/456 to fix compilation

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -43,6 +43,48 @@ jobs:
...
+        run: |
+          cmake -DCMAKE_BUILD_TYPE=Release .
+          make -j $(nproc)
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v2.2.1
+        with:
+          name: godot-cpp-linux-glibc2.23-x86_64-release
+          path: bin/libgodot-cpp.linux.release.64.a
+          if-no-files-found: error
```
Comment (Calinou) [inline .github/workflows/ci.yml:78]: No need to upload the Linux artifact for CMake, as we already have one built with SCons.

```suggestion
```

Comment (o01eg) [reply .github/workflows/ci.yml]: Removed it.

**Comment by o01eg** [comment]
Ping!

**Comment by o01eg** [comment]
Set minimum python version requirement

**Comment by o01eg** [comment]
I've rebased PR and updated project for recent version of Godot.

**Comment by bradc6** [comment]
@Calinou Is this something that can be merged in?

**Comment by Calinou** [comment]
> @Calinou Is this something that can be merged in?

I don't use CMake (and vnen doesn't either), so I don't know if the way this is implemented is correct.

**Comment by bradc6** [comment]
You want cmake lists files per sub section. Having the two is the standard way to do it

**Comment by bradc6** [comment]
@Calinou Would it be of any value for me to test it?

**Comment by Calinou** [comment]
> @Calinou Would it be of any value for me to test it for validation to merge it in?

CI checks have passed, so I guess we can test this in production (especially since this won't be cherry-picked to a stable branch).

**Review by Calinou — APPROVED** [review]
Thanks!

**Comment by o01eg** [comment]
Should I backport this PR to branchs for stable Godot version?

**Comment by Calinou** [comment]
> Should I backport this PR to branchs for stable Godot version?

Yes, on `3.x` specifically.

**Pull request merged**